theano/gpuarray/extra_ops.py

```diff
@@ -40,7 +39,6 @@ def get_params(self, node):
 
     def make_node(self, x):
         assert x.type.dtype == 'float32', "Only float32 supported for GpuCumSum"
-        x = as_gpuarray_variable(x, infer_context_name(x))
```

Comment:
I think it has to be done inside the `make_node`, because `make_node` can be called directly, or from another optimization, etc.
This is the way it is done all over `basic_ops.py`.